frontend/src/metabase/query_builder/actions/object-detail.js

```diff
@@ -56,13 +57,34 @@ export const followForeignKey = createThunkAction(
         return false;
       }
 
-      const newCard = startNewCard("query", card.dataset_query.database);
-
-      newCard.dataset_query.query["source-table"] = fk.origin.table.id;
-      newCard.dataset_query.query.filter = getFilterForFK(objectId, fk);
+      const databaseId = card.dataset_query.database;
+      const tableId = fk.origin.table.id;
+      const metadata = getMetadata(getState());
+      const metadataProvider = Lib.metadataProvider(databaseId, metadata);
+      const table = Lib.tableOrCardMetadata(metadataProvider, tableId);
+      const query = Lib.queryFromTableOrCardMetadata(metadataProvider, table);
+      const stageIndex = -1;
+      const column = Lib.fromLegacyColumn(query, stageIndex, fk.origin);
```

Comment:
I'm surprised it worked this way, but seems legit. Could you update `fromLegacyColumn` and include `Field` from `metabase-types/api` as well?

Reply:
234b538e2d0d356a2956784dd03582f6c7f69317